codegen/src/test/resources/software/amazon/awssdk/codegen/poet/auth/scheme/query-endpoint-auth-params-auth-scheme-endpoint-provider.java

```diff
@@ -0,0 +1,93 @@
+import software.amazon.awssdk.http.auth.AwsV4HttpSigner;
+import software.amazon.awssdk.http.auth.spi.AuthSchemeOption;
+import software.amazon.awssdk.services.query.auth.scheme.QueryAuthSchemeParams;
+import software.amazon.awssdk.services.query.auth.scheme.QueryAuthSchemeProvider;
+import software.amazon.awssdk.services.query.endpoints.QueryEndpointParams;
+import software.amazon.awssdk.services.query.endpoints.QueryEndpointProvider;
+import software.amazon.awssdk.utils.Validate;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+public final class DefaultQueryAuthSchemeProvider implements QueryAuthSchemeProvider {
+    private static final DefaultQueryAuthSchemeProvider DEFAULT = new DefaultQueryAuthSchemeProvider();
+
+    private static final QueryAuthSchemeProvider MODELED_RESOLVER = InternalModeledQueryAuthSchemeProvider.create();
+
+    private static final QueryEndpointProvider DELEGATE = QueryEndpointProvider.defaultProvider();
+
+    private DefaultQueryAuthSchemeProvider() {
+    }
+
+    public static QueryAuthSchemeProvider create() {
+        return DEFAULT;
+    }
+
+    @Override
+    public List<AuthSchemeOption> resolveAuthScheme(QueryAuthSchemeParams params) {
+        QueryEndpointParams endpointParameters = QueryEndpointParams.builder().defaultTrueParam(params.defaultTrueParam())
+                .defaultStringParam(params.defaultStringParam()).deprecatedParam(params.deprecatedParam())
+                .booleanContextParam(params.booleanContextParam()).stringContextParam(params.stringContextParam())
+                .operationContextParam(params.operationContextParam()).build();
+        Endpoint endpoint = DELEGATE.resolveEndpoint(endpointParameters).join();
+        List<EndpointAuthScheme> authSchemes = endpoint.attribute(AwsEndpointAttribute.AUTH_SCHEMES);
+        if (authSchemes == null) {
+            return MODELED_RESOLVER.resolveAuthScheme(params);
+        }
+        List<AuthSchemeOption> options = new ArrayList<>();
+        for (EndpointAuthScheme authScheme : authSchemes) {
+            String name = authScheme.name();
+            switch (name) {
+            case "sigv4":
+                SigV4AuthScheme sigv4AuthScheme = Validate.isInstanceOf(SigV4AuthScheme.class, authScheme,
+                        "Expecting auth scheme of class SigV4AuthScheme, got instead object of class %s", authScheme.getClass()
+                                .getName());
+                options.add(AuthSchemeOption
+                        .builder()
+                        .schemeId("aws.auth#sigv4")
+                        .putSignerProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, sigv4AuthScheme.signingName())
+                        .putSignerProperty(AwsV4HttpSigner.REGION_NAME, sigv4AuthScheme.signingRegion())
+                        .putSignerProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE,
+                                negateOrDefault(sigv4AuthScheme.disableDoubleEncoding())).build());
```

Comment:
sigv4AuthScheme.disableDoubleEncoding() is a `b`oolean, so just `!sigv4AuthScheme.disableDoubleEncoding()` here